src/meshes/abstract_tree.jl

```diff
@@ -387,13 +387,17 @@ function refine!(t::AbstractTree, cell_ids,
     return refined_original_cells
 end
 
+@inline function coordinate_min_max_check(coordinates_min, coordinates_max)
+    for dim in eachindex(coordinates_min)
+        @assert coordinates_min[dim]<coordinates_max[dim] "coordinates_min[$dim] must be smaller than coordinates_min[$dim]!"
+    end
+end
+
```

Comment:
```suggestion

coordinate_min_max_check(::Nothing, ::Nothing) = nothing
```
Would it make sense to add such a method to avoid the `nothing` check for `coordinates_min` and `coordinates_max` in the t8code and p4est meshes? This way we don't need to distinguish between the different mesh types whether or not we need the additional `if coordinates_min !== nothing && coordinates_max !== nothing` clause.

Reply:
Yeah looks reasonable to me. Would add a comment why we need this here though

Reply:
I added a comment